search/includes/classes/class-queue.php

```diff
@@ -288,6 +288,16 @@ public function count_jobs_due_now( $object_type = 'post' ) {
 		);
 	}
 
+	public function get_total_queue_size() {
```

Comment:
We already have a `count_jobs()` function, maybe we can alter that to support all object types (currently defaults to `post`, but if we allow `null` or `any`, we can conditionally alter the query to return all. Same with the status.

Reply:
Or, since our stats are per-index, which is technically per-type, we could count each type individually...hmm...

Reply:
Would it be useful to track the queue statuses individually?

Reply:
I'll make `count_jobs` handle 'all' statuses and 'all' object types.

Would probably be a good idea to pass in the object type when setting the stat just so we don't set both user and post queue gauges to the same value.